doc/man1/openssl-verification-options.pod

```diff
@@ -18,31 +18,56 @@ L<X509_verify_cert(3)> verify X.509 certificates. The details of how each
 command handles errors are documented on the specific command page.
 
 Certificate verification is a complicated process, consisting of
-a number of separate steps that are detailed in the following paragraphs.
+a number of separate steps that are detailed in the following sections.
+
+=head2 Certification Path Building
 
 First, a certificate chain is built up starting from the target certificate
 and typically ending in a self-signed "root" CA certificate.
 It is an error if the whole chain cannot be built up
 unless the B<-partial_chain> option is given.
+
 The chain is built up iteratively, looking up in turn
-the certificate of the signer ("issuer") of the current certificate.
+a currently valid (non-expired) certificate with suitable key usage that
+matches as an issuer of the current "subject" certificate as described below.
+In each iteration the first certificate found this way is taken.
+For efficiency, no backtracking is performed, thus
+any further candidate issuer certificates that would match equally are ignored.
+
 If a certificate is found that appears to be its own issuer
 it is assumed to be the self-signed root, which must be trusted.
 
-The process of looking up the issuer's certificate itself involves a number
-of steps.
-All available certificates with a subject name that matches the issuer
-name of the current certificate are subject to further tests.
-The relevant authority key identifier components of the current certificate
-(if present) must match the subject key identifier (if present)
-and issuer and serial number of the candidate issuer certificate.
+A candidate issuer certificate matches a subject certificate
+if all of the following conditions hold:
+
+=over 4
+
+=item * Its subject name equals the issuer name of the subject certificate.
+
+=item * If the subject certificate has an authority key identifier extension,
+each of its sub-fields equals the corresponding subject key identifier, serial
+number, and issuer field of the candidate issuer certificate, as far as present.
```

Comment:
The "as far as present" wording reads strangely to me, though I'm not sure I have an alternative wording that is clearly better.  Perhaps something like "each of the sub-fields that is present equals the corresponding [...]"?

Reply:
Hmm, this suggested formulation does not cover the case that the fields may not be present in the candidate issuer certificate.
I've replaced the "as far as present" by the more clear "as far as the respective fields are present in both certificates".